src/hooks/useGridDimensions.ts

```diff
@@ -17,26 +17,20 @@ export function useGridDimensions(): [
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     if (ResizeObserver == null) return;
 
-    function saveDimensions() {
-      // Get dimensions without scrollbars.
-      // The dimensions given by the callback entries in Firefox do not substract the scrollbar sizes.
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=1733042
-      const { clientWidth, clientHeight } = gridRef.current!;
+    const resizeObserver = new ResizeObserver((entries) => {
+      const size = entries[0].contentBoxSize[0];
       // TODO: remove once fixed upstream
       // we reduce width by 1px here to avoid layout issues in Chrome
       // https://bugs.chromium.org/p/chromium/issues/detail?id=1206298
-      setGridWidth(clientWidth - (devicePixelRatio % 1 === 0 ? 0 : 1));
-      setGridHeight(clientHeight);
-    }
-
-    saveDimensions();
```

Comment:
We no longer initialize the value. Let's revisit in the react 18 PR

Reply:
Do we not care about this anymore? We could still get `clientWidth`/`clientHeight` early.
https://github.com/adazzle/react-data-grid/commit/32a20f43aac88f56ed06e47fc9c0f6e8295002c8

Reply:
I can add it back but the value is not the same. Check the above comment

Reply:
Can you compare with `grid.getBoundingClientRect()`?

Reply:
`getBoundingClientRect()` doesn't remove the scrollbar width, try
`$0.getBoundingClientRect().width - ($0.offsetWidth - $0.clientWidth)`

Reply:
Added the the above logic back. It is however using `width/height` instead of `inlineSize/blockSize` which okay for now I gues